stan/math/rev/fun/sum.hpp

```diff
@@ -1,11 +1,14 @@
 #ifndef STAN_MATH_REV_FUN_SUM_HPP
 #define STAN_MATH_REV_FUN_SUM_HPP
 
+#include <stan/math/prim/fun/Eigen.hpp>
 #include <stan/math/rev/core.hpp>
 #include <stan/math/rev/meta.hpp>
+#include <stan/math/rev/functor/arena_matrix.hpp>
```

Comment:
This should be in `core` not `functor`. Mind making the change here? Otherwise make an issue and we'll do it somewhere else.

Reply:
Should it? It is not a feature, it is impossible to make rev implementations without. I guess it is more like a useful helper, like `reverse_pass_callback`.

Reply:
Oh. Good point. I'd always assumed what went in the 'functor' folder was anything that took in another function. But C++ functors are just objects that act like functors, and I don't really know what the math definition of functor means, so now I'm confused.

I just put things in core things that we use to make the rest of the autodiff work. This is one of those things. I'd probably put reverse_pass_callback in there too. If it seems confusing, let's just leave it where it is and bring this up at the next math meeting if anyone remembers.

Reply:
Hmm, I guess I just assumed everything that is not directly a function and not essential for the rest to go into functor (Arena matrices are not really functors. Same for many other things in functor). It might be a good idea to discuss folder structure in general on the meeting.

Reply:
imo I would like this to be in core and can move it over in this PR. Mostly on the basis that we will be using it in a lot of places and so conceptually it's sort of a core part of the autodiff

Reply:
tbh I'd rather actually move `reverse_pass_callback` to core as well. It's a weird semantic but I think of `core` as like "This will be used everywhere to implement reverse mode" where `functor` is more for operating over those implementations. `reverse_pass_callback` also only relies on core

Reply:
Oh yeah I agree with `reverse_pass_callback`. Can move here or later or whenever.